routers/auth.py

```diff
@@ -0,0 +1,189 @@
+import logging
+from contextlib import asynccontextmanager
+from datetime import datetime, timedelta
+
+from database_config import async_local_session, local_session
+from fastapi import APIRouter, Depends, HTTPException
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from jose import JWTError, jwt
+from models import User
+from passlib.context import CryptContext
+from pydantic import BaseModel, Field
+from sqlalchemy import select
+from starlette import status
+
+auth_router = APIRouter(
+    prefix="/auth", tags=["Auth"], responses={401: {"user": "user_not_authenticated"}}
+)
+logging.basicConfig(level=logging.DEBUG, filename="logs.txt")
+
+
+@asynccontextmanager
+async def get_db():
+    """
+    Calling this generator function will yield the DB object that can directly be used to query database.\n
+    What this async context manager does is close the session when our code stops using it.\n
+
+    """
+    try:
+        session = async_local_session
+        async with session() as db:
+            yield db
+    except Exception as e:
+        await db.rollback()
+    finally:
+        await db.close()
+
+
+def get_sync_db():
+    """
+    Calling this generator function will yield the DB object that can directly be used to query database.\n
+    """
+    try:
+        db = local_session()
+        yield db
+    except Exception as e:
+        db.rollback()
+    finally:
+        db.close()
+
+
+# CryptContext the object of the scheme passed i.e bcrypt that can be used
+# to hash passward before add them to db
+bcrypt_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+oauth_bearer = OAuth2PasswordBearer(tokenUrl="/login")
+
+SECRET_KEY = "my_super_secret_key"
```

Comment:
A good practice is to store private keys in an environment file or in a configuration file.